sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PropertiesShouldBePreferred.cs

```diff
@@ -68,31 +71,31 @@ protected override void Initialize(SonarAnalysisContext context)
         }
 
         private bool IsPropertyCanditate(IMethodSymbol method)
-        {
-            if (method.IsConstructor() ||
-                method.MethodKind == MethodKind.PropertyGet ||
-                method.IsOverride ||
-                method.GetInterfaceMember() != null ||
-                method.IsAsync ||
-                method.ReturnType.OriginalDefinition.IsAny(KnownType.SystemTasks))
-            {
-                return false;
-            }
-
-            return method.Parameters.Length == 0 &&
-                   !method.ReturnsVoid &&
-                   !method.ReturnType.Is(TypeKind.Array) &&
-                   method.Name != "GetEnumerator" &&
-                   method.Name != "GetAwaiter" &&
-                   NameStartsWithGet(method);
-        }
+            => method.Parameters.Length == 0
```

Comment:
While looking at this method, there are 10 conditions with a code smell. And 8 of them are negative. And there's no nice way out of it.

Can this be improved somehow? While I usually prefer arrow methods, in this case:
- changing `return && ! ...` to `if || ... return false` would remove some noise
- Return type is checked on 3 lines => could be extracted

Reply:
I though about grouping them in logical subsets, describing their intent. Ill try to improve it.

Reply:
Nice!